src/rpdk/core/jsonutils/utils.py

```diff
@@ -138,11 +140,17 @@ def schema_merge(target, src, path):  # noqa: C901
             try:
                 target[key] = schema_merge(target_schema, src_schema, next_path)
             except TypeError:
-                if key == "type":
-                    if isinstance(target_schema, list):
-                        target_schema.append(src_schema)
-                        continue
-                    target[key] = [target_schema, src_schema]
+                if key in (TYPE, REF):  # combining multiple $ref and types
+                    if isinstance(
+                        target.get(TYPE), list
+                    ):  # check not to add duplicates
+
+                        if src_schema not in target[TYPE]:
+                            target[TYPE].append(src_schema)
```

Comment:
can we use set union or something here to cut down on if statements and get rid of that pylint disable